dpc/src/virtual_machine/amount.rs

```diff
@@ -106,19 +108,59 @@ impl AleoAmount {
     }
 }
 
-impl Sum for AleoAmount {
-    fn sum<I: Iterator<Item = AleoAmount>>(iter: I) -> AleoAmount {
-        iter.fold(AleoAmount::ZERO, |a, b| a.add(b))
+impl Add<Self> for Amount {
```

Comment:
Should we also add an `impl Add<Result<Self>> for Amount` that simply propagates the error? Same for `Sub`.

Reply:
Do you mean something like this?
```
impl Add<Self> for Amount {
    type Output = Result<Amount>;

    fn add(self, other: Self) -> Self::Output {
        self.add(other)
    }
}
```

I believe `impl Add<Result<Self>> for Amount` means adding `Amount + Result<Amount>`

Reply:
I thought about setting, `type Output = Result<Amount>;` but ran into issues where we handle Amount arithmetic in iterators/maps/`sum()` and functions with no error handler. 

For simplicity, I went with the current `panic` approach. Open to other suggestions or potentially making all the functions handling Amount arithmetic require error handlers (if that's desired).